newIDE/app/src/CommandPalette/CommandManager.js

```diff
@@ -0,0 +1,43 @@
+// @flow
+type CmdHandler = () => void | Promise<void>;
+
+export type GlobalCommand = {|
+  displayText: string,
+  enabled: boolean,
+  handler: CmdHandler,
+|};
+
+export type UIGlobalCommand = {|
+  name: string,
+  ...GlobalCommand,
+|};
+
+export default class CommandManager {
+  globalCommands: { [string]: GlobalCommand };
+
+  constructor() {
+    this.globalCommands = {};
+    console.warn('Initialised command manager');
+  }
+
+  registerGlobal = (cmdName: string, cmdOpts: GlobalCommand) => {
```

Comment:
Let's give a more precise name. What are we registering? A command :)
`registerCommand`.